### ISSUE TITLE
Tests use user configuration instead of default configuration

### ISSUE THREAD

**Issue by ReimarBauer** [Open-MSS/MSS#471]
**[Original report](https://bitbucket.org/wxmetvis/mss/issue/471) by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

The user defined configuration influences test cases and can introduce errors. In one example, the user configuration contained erroneous values for the vertical range of the side view that were introduced on an experimental branch. These causes the tests in the stable branch to fail later on.

‌

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

same situation for the mss-core.conf which has maybe incompatible qt cached values

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

the code base needs a review, usually   
  
export MSS\_CONFIG\_PATH="/tmp/tmp78h01gas/.config/mss"

should be enough to set this path, but it is ignored.

‌

‌

```
# set "MSS_CONFIG_PATH"

os.environ["MSS_CONFIG_PATH"] = os.path.join(tempfile.mkdtemp(), ".config", "mss")
```

‌

**Comment by ReimarBauer** [comment]
**Original comment by Joern Ungermann (Bitbucket: [joernu76](https://bitbucket.org/joernu76), GitHub: [joernu76](https://github.com/joernu76)).**

----------------------------------------

There is some code hidden away in `mslib/_tests/constants.py` that changes the environment for testing. That should prevent the user’s config from interfering with the tests. This seems to work on my laptop. Maybe you had a different error with the config?

BTW, I do not think that the current solution is very robust. I think, this should be moved to conftest.py where it can be applied before importing _any_ mslib code.

**Comment by ReimarBauer** [comment]
partly adressed with https://github.com/Open-MSS/MSS/pull/663

**Comment by ReimarBauer** [comment]
the missing part is the msui-core.conf file this needs to be used on testing also from the tmp dir.

**Comment by ReimarBauer** [comment]
fixed by #1645
